src/language/syntax/rules/profile/profile.ts

```diff
@@ -509,11 +507,12 @@ const PROFILE_VERSION = SyntaxRule.identifier('version')
         label?: string;
       } & LocationInfo
     >(version => {
-      const parseVersionResult = parseVersion(version.data.string);
-      if (parseVersionResult.kind !== 'parsed') {
-        return { kind: 'nomatch' };
-      }
-      const parsedVersion = parseVersionResult.value;
+      const parsedVersion = VersionRange.fromString(version.data.string);
+      //TODO: nomatch on version err?
+
+      // if (parseVersionResult.kind !== 'parsed') {
+      //   return { kind: 'nomatch' };
+      // }
```

Comment:
Yes, please _do not_ throw in there, this is designed to return nomatch if the syntax is bad. Throw is only for internal parser errors.